examples/multipart-form/main.rs

```diff
@@ -0,0 +1,73 @@
+#![feature(async_await, futures_api)]
+
+use multipart::server::Multipart;
+use std::io::{Cursor, Read};
+use tide::{
+    body,
+    head::{NamedComponent, NamedHeader},
+    App,
+};
+
+struct ContentType(String);
+
+impl NamedComponent for ContentType {
+    const NAME: &'static str = "content-type";
+}
+
+impl std::str::FromStr for ContentType {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(ContentType(s.to_string()))
+    }
+}
+
+async fn form_auto(file: body::MultipartFile) -> String {
+    let content = String::from_utf8(file.0).unwrap_or("Error".to_string());
+    println!("[form_auto]: file content: {}", content);
+    content
+}
+
+async fn form_manual(
+    NamedHeader(ct): NamedHeader<ContentType>,
+    multipart_body: body::Bytes,
+) -> String {
+    // https://stackoverflow.com/questions/43424982/how-to-parse-multipart-forms-using-abonander-multipart-with-rocket
+    let ContentType(content_type) = ct;
+    println!("content_type: {}", content_type);
+
+    let idx = content_type.find("boundary=").expect("no boundary");
+    let boundary = &content_type[(idx + "boundary=".len())..];
+    println!("bounday={}", boundary);
+
+    let mut mp = Multipart::with_body(Cursor::new(multipart_body.0), boundary);
+    let mut file = None;
+    mp.foreach_entry(|mut entry| match entry.headers.name.as_str() {
+        "file" => {
+            let mut vec = Vec::new();
+            entry.data.read_to_end(&mut vec).expect("can't read");
+            file = Some(vec);
+        }
+        other => panic!("No known key {}", other),
+    })
+    .expect("Unable to iterate");
+
+    let file = file.expect("file not set");
+
+    let content = String::from_utf8(file).unwrap_or("Error".to_string());
+
+    println!("[form_manual] file content: {}", content);
+    content
+}
+
+fn main() {
+    let mut app = App::new(());
+
+    app.at("/form_auto").post(form_auto);
+    app.at("/form_manual").post(form_manual);
+
+    app.serve("127.0.0.1:7878");
+}
+
+// Test with:
+// curl -X POST   http://localhost:7878/form_manual -H 'Cache-Control: no-cache'   -H 'Postman-Token: e26f7bce-51b4-4d12-aba8-439ff8549bb9'   -H 'content-type: multipart/form-data'   -F file=@examples/multipart-form/test.txt
+// curl -X POST   http://localhost:7878/form_auto   -H 'Cache-Control: no-cache'   -H 'Postman-Token: e26f7bce-51b4-4d12-aba8-439ff8549bb9'   -H 'content-type: multipart/form-data'   -F file=@examples/multipart-form/test.txt
```

Comment:
Great idea adding these curl requests! One nit - I think we can remove the `postman-token` header :)